ledger-core/v2/virtual/execute/execute.go

```diff
@@ -178,14 +211,43 @@ func (s *SMExecute) stepTakeLock(ctx smachine.ExecutionContext) smachine.StateUp
 		return ctx.Sleep().ThenRepeat()
 	}
 
+	return ctx.Jump(s.stepGetObjectDescriptor)
+}
+
+func (s *SMExecute) stepGetObjectDescriptor(ctx smachine.ExecutionContext) smachine.StateUpdate {
+	var objectDescriptor descriptor.ObjectDescriptor
+
+	action := func(state *object.SharedState) {
+		objectDescriptor = state.Descriptor()
+	}
+
+	switch s.objectSharedState.Prepare(action).TryUse(ctx).GetDecision() {
```

Comment:
this block copypasted already 3 times, and I wanted to copypaste it again. Maybe we need helper here?

Reply:
Not a time to refactor it, it'll be better to do it on unordered path